diff --git a/src/bindings/python/flux/rpc.py b/src/bindings/python/flux/rpc.py
index 7af1fe7b..c19109e4 100644
--- a/src/bindings/python/flux/rpc.py
+++ b/src/bindings/python/flux/rpc.py
@@ -57,8 +57,12 @@ class RPC(WrapperPimpl):

     def get_str(self):
         j_str = ffi.new('char *[1]')
-        self.pimpl.get(j_str)
-        return ffi.string(j_str[0])
+        try:
+            self.pimpl.get(j_str)
+            return ffi.string(j_str[0])
+        except EnvironmentError as e:
+            errmsg = ffi.string(raw.flux_future_error_string(self.handle))
+            raise EnvironmentError(e.errno, errmsg)

     def get(self):
         return json.loads(self.get_str().decode('utf-8'))
